regex-assembly/942380.ra

```diff
@@ -3,23 +3,19 @@
 
 ##!+ i
 
-\bhaving\b\s+\d{1,10}\s*?[=<>]
-\bhaving\b\s+'[^=]{1,10}'\s*?[=<>]
-\bexecute\(
-\bexecute\s{1,5}[\w\.$]{1,5}\s{0,3}
-\bhaving\b ?\d{1,10} ?[=<>]+
-\bhaving\b ?[\'\"][^=]{1,10}[\'\" ?[=<>]+
-\bcreate\s+?table.{0,20}?\(
+\bhaving\s*\(?\s*(?:['"]?\s*\d+\s*['"]?|['"]\s*[^\d]*\s*['"]|true|false)\s*(?:[=<>+\-*/%^&!|]+|div|mod)
+\bexecute\s*\(
+\bexecute\b\s*`?\s*[\w\.$]+\s*`?
+\bcreate\b\s+\btable\b\s*`?\s*.+?\s*`?\s*\(
```

Comment:
Too complicated. Break it up into alternations, e.g.:
```suggestion
\bcreate\b\s+\btable\b[^(]+\(
\bcreate\b\s+\btable\b\s*`\s*[^\s`]+\s*`\s*\(
```

Reply:
@theseion But it's not the same. Your version will match also this:
`create table`
Mine is matching, at least, this:
`create table (`

I don't think this one can be splitted into parts.

Reply:
How do my proposed expressions match `create table`? Both require `\(`. `?` can always be written as an alternation (`|`), in this case
```
...(?:`|)...
```
And `[^\(]+` is just a simplification of the wildcards in the original expression. So yes, I'm pretty sure that's correct :).

Reply:
Yes. The first doesn't have the backtick enclosed part, the second always has backticks. The result would be (spaces added for readability):
```
\bcreate\b\s+\btable\b[^(]+\(  |  \bcreate\b\s+\btable\b\s*`\s*[^\s`]+\s*`\s*\(
```
Shortened:
```
\bcreate\b\s+\btable\b(?:[^(]+\(  |  \s*`\s*[^\s`]+\s*`\s*\()
```